### ISSUE TITLE
BadgeSymbol width not updated if defined as an `@State`

### ISSUE THREAD

**Issue by balloonio** [balloonio/introducing-swift-ui#1]
## TL;DR
Broken Commit: https://github.com/balloonio/introducing-swift-ui/commit/8bf1c3b7af0feb302581de9b89e4e270eafc0cbb
Working Commit: https://github.com/balloonio/introducing-swift-ui/commit/9ee72e0e5bb827f2bc39287f335cf0a8e022bed2

## Problem
With the broken commit, In `Badge.swift` preview, the `BadgeSymbol` is not rendered with the updated `width` from `GeometryReader`.  (See screenshot, the little triangle is with the size of the `@State private var width = 100` inside `BadgeSymbo.swift` without being updated by L27 `width = min(geometry.size.width, geometry.size.height)`
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/3913002/187333398-d8e198d2-16e0-418f-a8f1-5ddad8fcc763.png">
 
However, with the broken commit, the preview in `BadgeSymbol.swift` itself was correct.
<img width="1225" alt="image" src="https://user-images.githubusercontent.com/3913002/187333791-534f7ca3-d29a-4592-8bcc-3bb6ced51a71.png">

## Expected Behavior

I expect the `BadgeSymbol` to correctly show inside `Badge.swift`, like below.
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/3913002/187334037-905484a0-0baf-4299-869b-084f28bcb24d.png">

## Observation
Since the only difference between the working version and the broken version is to change 

```
    @State private var width : CGFloat = 100  // define state variable
...
    var body: some View {
        GeometryReader { geometry in
            
            Path { path in
                width = min(geometry.size.width, geometry.size.height)  // and update its value within body
...
```
to
```
    var body: some View {
        GeometryReader { geometry in
            Path { path in
                let width = min(geometry.size.width, geometry.size.height)  // directly create a constant within the body
```

There must be some issue with the `@State` usage here, but I don't understand what is wrong

**Comment by balloonio** [comment]
More discussion in https://www.reddit.com/r/SwiftUI/comments/x17cx7/state_variable_from_child_view_not_being/
A workaround is to use an `ObservableObject` as a wrapper
